### PR TITLE
Create index and show for book

### PR DESCRIPTION
**Pull request by fedecvz**
## Summary:
Create index and show for book

## Trello card:
https://trello.com/c/Qc9UK6oL/24-crear-rutas-index-y-show-para-book

### CODE REVIEW

File: app/controllers/api/v1/book_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class BookController < Api::V1::ApiController
+      def index
+        @books = Book.all
+        render json: @books
+      end
+
+      def show
+        @book = Book.find(params[:id])
```
Comment (juanisierra) [inline app/controllers/api/v1/book_controller.rb]: You should use [strong params](https://edgeapi.rubyonrails.org/classes/ActionController/StrongParameters.html)
Also, what happens if the book_id doesn't exist?

File: config/routes.rb [hunk 2]
```diff
@@ -1,4 +1,10 @@
 Rails.application.routes.draw do
+  namespace :api do
+    namespace :v1 do
+      get 'books', to: 'book#index'
```
Comment (juanisierra) [inline config/routes.rb]: Use routing resources so as to standardize URLS. Check this doc https://guides.rubyonrails.org/routing.html

File: app/controllers/api/v1/book_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class BookController < Api::V1::ApiController
```
Comment (juanisierra) [inline app/controllers/api/v1/book_controller.rb]: `BooksController`

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: app/serializers/api/v1/book_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module Api
+  module V1
+    class BookSerializer < ActiveModel::Serializer
```
Comment (juanisierra) [inline app/serializers/api/v1/book_serializer.rb:3]: Are you only returning id?

File: config/routes.rb [hunk 5]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
   namespace :api do
     namespace :v1 do
+      resources :books
```
Comment (juanisierra) [inline config/routes.rb]: You should only add the implemented methods, not all of them

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: app/serializers/api/v1/book_serializer.rb (hunk 4 above)
Comment (fedecvz) [reply app/serializers/api/v1/book_serializer.rb:3]: I already implemented it in the book-test-rent-index-show-paginate-test-book-index branch.

Comment (fedecvz) [reply app/serializers/api/v1/book_serializer.rb:3]: With attributes: :id, :genre, :author, :image, :title, :publisher, :year, :created_at, :updated_at

**Review by juanisierra — APPROVED** [review]

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/books_controller.rb [hunk 6]
```diff
@@ -6,13 +6,13 @@ def index
       end
 
       def show
-        render json: Book.find(books_params)
+        render json: Book.find(books_params['id'])
```
Comment (juanisierra) [inline app/controllers/api/v1/books_controller.rb]: You can use a symbol to access the hash instead of a string

**Review by juanisierra — APPROVED** [review]

**Pull request closed without merging**